whitesource/client.py

```diff
@@ -64,41 +64,47 @@ def save_project_tag(self, projectToken: str, key: str, value: str):
             json=body,
         )
 
-    def post_project(
+    async def ws_project(
         self,
         extra_whitesource_config: typing.Dict,
         file,
-        filename: str,
         project_name: str,
         requester_email: str,
+        length: int,
+        chunk_size: int,
+        ping_interval: int,
+        ping_timeout: int,
     ):
 
-        fields = {
-            'component': (f'{filename}.tar', file, 'application/zip'),
-            'wsConfig': json.dumps({
-                'apiKey': self.api_key,
-                'projectName': project_name,
-                'productToken': self.product_token,
-                'requesterEmail': requester_email,
-                'userKey': self.creds.user_key(),
-                'wssUrl': self.wss_endpoint,
-                'extraWsConfig': extra_whitesource_config,
-            })
+        meta_data = {
+            'chunkSize': chunk_size,
+            'length': length
         }
 
-        # add extra whitesource config
-        for key, value in extra_whitesource_config.items():
-            fields[key] = value
+        ws_config = {
+            'apiKey': self.api_key,
+            'extraWsConfig': extra_whitesource_config,
+            'productToken': self.product_token,
+            'projectName': project_name,
+            'requesterEmail': requester_email,
+            'userKey': self.creds.user_key(),
+            'wssUrl': self.wss_endpoint,
+        }
 
-        m = MultipartEncoder(
-            fields=fields,
-        )
-        return self.request(
-            method='POST',
-            url=self.routes.post_component(),
-            headers={'Content-Type': m.content_type},
-            data=m,
-        )
+        async with websockets.connect(
+            uri=self.routes.ws_component(),
+            ping_interval=ping_interval,
+            ping_timeout=ping_timeout,
+            ) as websocket:
+            await websocket.send(json.dumps(meta_data))
+            await websocket.send(json.dumps(ws_config))
+            while True:
+                chunk = file.read(chunk_size)
+                if not chunk:
+                    break
+                await websocket.send(chunk)
+
+            return await websocket.recv()
```

Comment:
what's the response (sorry, too lazy to lookup on server-side)? if it is again some machine-readable piece of json, it would be nice to deserialise it prior to returning